mm2src/coins_activation/src/utxo_activation/init_utxo_standard_activation.rs

```diff
@@ -61,31 +60,20 @@ impl InitStandaloneCoinActivationOps for UtxoStandardCoin {
         let crypto_ctx = CryptoCtx::from_ctx(&ctx)?;
         let priv_key_policy = priv_key_build_policy(&crypto_ctx, activation_request.priv_key_policy);
 
-        let (sync_status_loop_handle, maybe_sync_watcher) =
-            if coin_conf["enable_spv_proof"].as_bool().unwrap_or(false) && !activation_request.mode.is_native() {
-                let (sync_status_notifier, sync_watcher) = channel(1);
-                (
-                    Some(UtxoSyncStatusLoopHandle::new(sync_status_notifier)),
-                    Some(sync_watcher),
-                )
-            } else {
-                (None, None)
-            };
-
         let coin = UtxoArcBuilder::new(
             &ctx,
             &ticker,
             &coin_conf,
             activation_request,
             priv_key_policy,
-            sync_status_loop_handle,
             UtxoStandardCoin::from,
         )
         .build()
         .await
         .mm_err(|e| InitUtxoStandardError::from_build_err(e, ticker.clone()))?;
 
-        if let Some(mut sync_watcher) = maybe_sync_watcher {
+        if let Some(sync_watcher_mutex) = &coin.as_ref().block_headers_status_watcher {
```

Comment:
At one of the upcoming PRs we should move this into a common function that will be used for other coins. For example, we might be interesting to use block-headers storage for QTUM coin and even probably for BCH

Reply:
I agree. Added this comment to the checklist https://github.com/KomodoPlatform/atomicDEX-API/issues/1045#issue-971642359